smartsim/_core/control/controller.py

```diff
@@ -854,7 +851,7 @@ def reload_saved_db(self, checkpoint_file: str) -> Orchestrator:
             try:
                 for db_job, step in job_steps:
                     self._jobs.db_jobs[db_job.ename] = db_job
```

Comment:
It seems like this process of rehydrating tasks would be an operation that could be isolated from `reload_saved_db`. For example, the current implementation of the telemetry monitor would benefit from it.

- Q1. Should we extract the job management section into a useful new method?

The addition of `add_step_to_mapping_table` is ok but it seems odd that we only use it once and that we also allow direct access for reads (e.g. `steps.append(self._launcher.step_mapping[db_job.name]` in `_save_orchestra`). We also allow writes via `StepMapping.add` in [dragonLauncher.py](https://github.com/CrayLabs/SmartSim/pull/541/files#diff-3ad221198dd265c029606dd8fb1e0601d66edc2cefe9b7007cf010297381eceaR150-R158).

- Q2. Do we need `add_step_to_mapping_table` or can we re-use `StepMapping.add`?